localstack/services/lambda_/provider.py

```diff
@@ -1823,7 +1823,8 @@ def create_function_url_config(
         url_id = api_utils.generate_random_url_id()
 
         host_definition = localstack_host(
-            use_localhost_cloud=True, custom_port=config.EDGE_PORT_HTTP or config.EDGE_PORT
+            use_localhost_cloud=True,
+            custom_port=config.EDGE_PORT_HTTP or config.GATEWAY_LISTEN[0].port,
```

Comment:
I would remove the EDGE_PORT_HTTP here, makes no sense I think, as those values are likely the same.

Reply:
I thought about batching breaking changes in a separate PR that removes `EDGE_PORT_HTTP`, `EDGE_BIND_HOST`, and write-access of `EDGE_PORT` localstack-wide.
I can also sneak it in here if you think it doesn't affect anyone 🤷